pandas/plotting/_matplotlib/core.py

```diff
@@ -482,6 +488,11 @@ def _adorn_subplots(self):
             if self.xlim is not None:
                 ax.set_xlim(self.xlim)
 
+            # GH9093, currently Pandas does not show ylabel, so if users provide
+            # ylabel will set it as ylabel in the plot.
+            if self.ylabel is not None:
+                ax.set_ylabel(self.ylabel)
```

Comment:
Is this consistent with `xlabel`, which seems to `pprint` the value?

May be I'm wrong, but feels like `df.plot(xlabel=a_complex_structure)` will make `a_complex_structure` (may be a dict of lists) look "nice". While `df.plot(ylabel=a_complex_structure)` may fail, or just convert to string with `str` instead of `pprint_thing`.

Would be probably good to add a test in the parametrization to make sure this works as expected.

Reply:
good call! i think matplotlib's `ax.set_ylabel` already could make the complex structure case look nice by converting to string automatically without using `pprint_thing`, e.g. inputs like `[1, 2, 3]` will be shown as `'[1, 2, 3]'` same in matplotlib plot. 

But i think it might be indeed nicer to be consistent with `xlabel` so I added `pprint_thing` for `ylabel` as well, and add the test case in parametrization! thanks!